src/networking.c

```diff
@@ -419,8 +396,21 @@ void _addReplyToBufferOrList(client *c, const char *s, size_t len) {
         return;
     }
 
-    size_t reply_len = _addReplyToBuffer(c,s,len);
-    if (len > reply_len) _addReplyProtoToList(c,c->reply,s+reply_len,len-reply_len);
+    /* We update the buffer peak always */
+    const size_t available = c->buf_usable_size - c->bufpos;
+
+    /* If there already are entries in the reply list or the added length surpasses the buffer,
+     * we cannot add anything more to the static buffer. */
+    const int out_buffer_boundary = len > available;
+    if (listLength(c->reply) > 0 || out_buffer_boundary)
```

Comment:
If we run out of c->buf space, we still need to write as much data as possible to it before we write to the reply list.